source/agora/consensus/SlashPolicy.d

```diff
@@ -267,6 +268,13 @@ public class SlashPolicy
             assert(0);
         }
 
+        auto enroll_index = this.enroll_man.getIndexOfEnrollment();
+        if (enroll_index != ulong.max && !missing_validators.find(enroll_index).empty)
+        {
+            log.fatal("The node is slashing itself.");
+            assert(0);
+        }
```

Comment:
Isn't this `checkSelfSlashing` ?

Reply:
Yes, it is.
Changed it to call `checkSelfSlashing` and moved the calling into the ledger.